# Reproduce the analysis of the Hard to Employ program in NY

## Null hypothesis: The % of inmates who was convicted of felony is the same or larger than those who participated in the program as for the control groups. 
## significance level , $alpha = 0.05$

$H_0$: $P_0$ >= $P_1$

$H_a$: $P_0$ < $P_1$



    



In [30]:

import os
import sys
import numpy as np
import pylab as pl


import json
import os
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [22]:
alpha = 0.05
P_0 = 10.0 * 0.01
P_1 = 11.7 * 0.01

if P_0 >= P_1:
    print 'the Null holds'
else:
    print 'We must assess the statistical significance'

n_0 = 568
n_1 = 409

N_0 = P_0 * n_0
N_1 = P_1 * n_1

We must assess the statistical significance


In [23]:
#define the sample proportion first
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print sp

0.107116683726


In [24]:
p = lambda p0, p1, n0, n1: (p0 * n0 + p1 * n1) / (n0 + n1)
#standard error
se = lambda p, n0, n1: np.sqrt(p * (1 - p) * (1.0 / n0 + 1.0 / n1)) 

In [26]:
zscore = lambda p0, p1, s : (p0 - p1) / s
z_y = zscore(P_1, P_0, se(p(P_0, P_1, n_0, n_1), n_0, n_1))
print (z_y)

0.84764020522


In [27]:
p_y = 1-0.8023 #after looking up the value in the table
def report_result(p,a):
    print ('Is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('the Null hypothesis is {}'.format( 'rejected' if p < a  else 'not rejected') )

    
report_result(p_y,alpha)

Is the p value 0.20 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


## fill in the contingency table below

|convicted of a fellony | yes | no |
|:----------------------| ---:|:-:|
|test sample|0.10*568|0.90*568	|568|
|control sample|0.117*409|0.883*4.09|	409|
|total|	104.653	|872.347|977|

In [32]:
from evalChisq import evalChisq

In [33]:
Ntot = 977
sample_values = np.array([[0.1 * 568, 0.9 * 568], [0.117 * 409, 0.883 * 409]])

In [34]:
print evalChisq(sample_values)

0.718493917505


### DOF =1
### According to the statistical table, while the siginificant level = 0.05, the value of Chi-square is 3.84. However, what we count is 0.718 which is much smaller than the 3.84. Therefore, the null hypothesis could not be rejected. 